### DISCUSSION TITLE
How to do interdependent stores lazyLoad

### DISCUSSION THREAD

**Discussion by aeischeid** [ai/nanostores#292] [General]
I have a store that is dependent on data from another store as it mounts / initializes. Using `computed` might be a solution, but subscribing in this case felt a bit overkill, and the store item I would subscribe to changes enough that I didn't want to do that.

So wondering if there is a recommended way to  do this sort of thing, similar to computed maybe but only runs once? 

For my specific case here was one solution I came up which feels a little hacky, but seems to work okay so far 

``` ts
onMount(activeReplyId, async () => {
	let $user = user.get();

	// There is a race condition potential where this onMount could be called before UserStore has been set
	// subscribing to user might work, but we know it will be there pretty quick
	// so using a sleepy while loop feels simpler than subscribing and unsubscribing just to finish this initialization

	/* Returns a Promise that resolves after "ms" Milliseconds */
	const sleep = (ms: number) => new Promise((res) => setTimeout(res, ms));

	let n = 0; // to avoid infinite load loop 
	while (!$user.organization_id && n < 20) {
		await sleep(200);
		$user = user.get();
		n++;
	}

	const replyId = $user.replyIds[0];
	activeReplyId.set(replyId);
	getCampaignData(replyId);
});

**Comment by ai** [comment]
I personally think `computed` is the best here, since it is so cool if web app is ready to user changes (sign out, re-sign in).

But if you want to keep this approach, here is a better way without `sleep()`:

```js
onMount(activeReplyId, async () => {
  let unbind = user.subscribe($user => {
    if ($user.organization_id) {
      unbind();
      const replyId = $user.replyIds[0];
      activeReplyId.set(replyId);
      getCampaignData(replyId);
    }
  })
  return unbind;
});
```
